tests/monitors/df/df_test.py

```diff
@@ -1,67 +1,81 @@
 import pytest
 
-from tests.helpers.agent import Agent
-from tests.helpers.assertions import has_log_message
 from tests.helpers.metadata import Metadata
-from tests.helpers.util import wait_for
+from tests.helpers.verify import verify_custom, verify_included_metrics, verify_all_metrics
 
 pytestmark = [pytest.mark.collectd, pytest.mark.df, pytest.mark.monitor_without_endpoints]
 
 METADATA = Metadata.from_package("collectd/df")
 
 
-def test_df():
-    with Agent.run(
-        """
+def test_df_included_metrics():
+    agent_config = """
         monitors:
           - type: collectd/df
             hostFSPath: /
         """
-    ) as agent:
-        _ = (
-            wait_for(lambda: set(agent.fake_services.datapoints_by_metric) == METADATA.included_metrics),
-            "timed out waiting for metrics and/or dimensions!",
-        )
-        assert set(agent.fake_services.datapoints_by_metric) == METADATA.included_metrics
-        assert not has_log_message(agent.output.lower(), "error"), "error found in agent output!"
+    verify_included_metrics(agent_config, METADATA)
 
 
 def test_df_extra_metrics():
-    expected_metrics = METADATA.included_metrics | {"df_complex.reserved"}
-    with Agent.run(
-        """
+    df_complex_reserved, df_inodes_reserved = "df_complex.reserved", "df_inodes.reserved"
+    expected_metrics = METADATA.included_metrics | {df_complex_reserved, df_inodes_reserved}
+    agent_config = f"""
         monitors:
           - type: collectd/df
             hostFSPath: /
             extraMetrics:
-            - df_complex.reserved
+            - {df_complex_reserved}
+            - {df_inodes_reserved}
         """
-    ) as agent:
-        _ = (
-            wait_for(lambda: set(agent.fake_services.datapoints_by_metric) == expected_metrics),
-            "timed out waiting for metrics and/or dimensions!",
-        )
-        assert set(agent.fake_services.datapoints_by_metric) == expected_metrics
-        assert not has_log_message(agent.output.lower(), "error"), "error found in agent output!"
+    verify_custom(agent_config, expected_metrics)
 
 
-def test_df_extra_metrics_all():
-    expected_metrics = METADATA.all_metrics
-    with Agent.run(
+def test_df_report_inodes_flag():
+    expected_metrics = METADATA.included_metrics | {"df_inodes.free", "df_inodes.reserved", "df_inodes.used"}
```

Comment:
Can use `METADATA.metrics_by_group["report-inodes"]` (though there's something to be said for testing that the groups are correct in the metadata by explicitly listing the metrics). @keitwb what do you think?

Reply:
Yeah nothing wrong with enumerating them explicitly as an extra thorough check.